lib/formElement/Date.js

```diff
@@ -0,0 +1,27 @@
+( function ( mw, $, undefined ) {
+	mw.ext.forms.formElement.Date = function() {};
+
+	OO.inheritClass( mw.ext.forms.formElement.Date, mw.ext.forms.formElement.InputFormElement );
+
+	mw.ext.forms.formElement.Date.prototype.getElementConfig = function() {
+		var config = mw.ext.forms.formElement.Date.parent.prototype.getElementConfigInternal.call( this );
+
+		config.value.type = 'date';
```

Comment:
You could add a control here (for the editor) to specify the `inputFormat` (and/or any other specific configs for this widget)

Reply:
Actually, there is nothing I need atm. I removed getElementConfig completely